catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/Highlight.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.operation;
+
+/**
+ * A Highlight represents a specific range of characters in an attribute that matched a query. The
+ * begin and end indices are referenced to the specific attribute value. The value index refers to
+ * multi-valued fields and indicates which value the highlight applies to.
+ */
+public interface Highlight {
```

Comment:
✏️ I wouldn't hold the PR up for this, but we should probably mark this as an experimental interface